en/docs/install-and-setup/upgrading-wso2-api-manager/260-to-410/upgrading-from-260-to-410.md

```diff
@@ -0,0 +1,1304 @@
+
+2. Copy the `<AM_MIGRATION_CLIENT_HOME>/migration-resources` to the `<API-M_4.1.0_HOME>` directory.
+
+3. Copy the org.wso2.carbon.am.migration-4.1.0.x.jar file in the `<AM_MIGRATION_CLIENT_HOME>/dropins` directory into the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+4.  Start the API-M server to migrate the API-M components as follows.
+
+    ``` tab="Linux / Mac OS"
+    sh api-manager.sh -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+    ``` tab="Windows"
+    api-manager.bat -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+5.  Shutdown the API-M server.
+
+    -   Remove the `org.wso2.carbon.apimgt.migrate.client-4.1.0.x.jar` file, which is in the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+    -   Remove the `migration-resources` directory, which is in the `<API-M_4.1.0_HOME>` directory.
+
+    !!! note
+
+        Make sure you have copied the tenants userstores if you have configured them in API-M 2.6.0
+
+### Step 5: Re-Index the API Manager artifacts
+
+1. To re-index the API artifacts in the registry, Add the following configuration into the `<API-M_4.1.0_HOME>/repository/conf/deployment.toml` file.
+
+    ```
+    [indexing]
+    re_indexing = 1
+    ```
+
+   Note that you need to increase the value of `re_indexing` by one each time you need to re-index.
+
+
+2. If the `<API-M_4.1.0_HOME>/solr` directory exists, take a backup and thereafter delete it.
+
+3. Restart the server.
+
+    ```tab="Linux / Mac OS"
+        sh api-manager.sh
+    ```
+
+    ```tab="Windows"
+        api-manager.bat
+    ```
+
+!!! important
+    If you use a clustered/distributed API Manager setup, do the above change in deployment.toml of Publisher and Devportal nodes. Make sure to keep a delay between nodes to execute this step to re-index each node, as the database can experience a large load.
+
+!!! note
+    Note that it takes a considerable amount of time for the API Manager to re-index the artifacts, depending on the API and the number of tenants.
+
+### Step 6 - Restart the WSO2 API-M 4.1.0 server
+
+1.  Restart the WSO2 API-M server.
+
+    ```tab="Linux / Mac OS"
+    sh api-manager.sh
+    ```
+
+    ```tab="Windows"
+    api-manager.bat
+    ```
+
+This concludes the upgrade process.
+
+!!! tip
+    The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
+
+!!! note
+    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
```

Comment:
API-M 4.1.0

Reply:
fixed in https://github.com/wso2/docs-apim/pull/5985/commits/8fc2d9f27c21734726ffc418226b824dd2c9132c#diff-b598cabec9a35d16c1c419398bd373cca84f31896a22a34bc24448b9606d99b0R1282